launcher/launcher.go

```diff
@@ -81,10 +81,12 @@ func (l *launcher) Run(b *beat.Beat) error {
 	if err := b.Manager.Start(); err != nil {
 		return err
 	}
-	defer b.Manager.Stop()
 
 	// Wait for Fleet-side reconfiguration only if beater is running in Agent-managed mode.
 	if b.Manager.Enabled() {
+
+		defer b.Manager.Stop()
+
```

Comment:
this needs to run only when we're in managed mode otherwise it will call our `Stop()` function again which will result in a panic.